extensions/Worker.Extensions.Tables/src/TableConverter.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            if (!IsTableExtension(modelBindingData))
+            {
+                return ConversionResult.Unhandled();
+            }
+            try
+            {
+                Dictionary<string, object> content = GetBindingDataContent(modelBindingData);
+                var result = await ConvertModelBindingDataAsync(content, context.TargetType, modelBindingData);
+
+                if (result is not null)
+                {
+                    return ConversionResult.Success(result);
+                }
+            }
+            catch (Exception ex)
+            {
+                return ConversionResult.Failed(ex);
+            }
+
+            return ConversionResult.Unhandled();
+        }
+
+        internal bool IsTableExtension(ModelBindingData bindingData)
+        {
+            if (bindingData?.Source is not Constants.TablesExtensionName)
+            {
+                _logger.LogTrace("Source '{source}' is not supported by {converter}", bindingData?.Source, nameof(TableConverter));
+                return false;
+            }
+
+            return true;
+        }
+
+        public virtual async Task<object?> ConvertModelBindingDataAsync(IDictionary<string, object> content, Type targetType, ModelBindingData bindingData)
+        {
+            content.TryGetValue(Constants.Connection, out var connection);
+            content.TryGetValue(Constants.TableName, out var tableName);
+            content.TryGetValue(Constants.PartitionKey, out var partitionKey);
+            content.TryGetValue(Constants.RowKey, out var rowKey);
+            content.TryGetValue(Constants.Take, out var take);
+            content.TryGetValue(Constants.Filter, out var filter);
+
+            if (string.IsNullOrEmpty(tableName?.ToString()))
+            {
+                throw new ArgumentNullException("'TableName' cannot be null or empty");
+            }
+
+            return await ToTargetTypeAsync(targetType, connection?.ToString() ?? null, tableName!.ToString(), partitionKey?.ToString() ?? null, rowKey?.ToString() ?? null, Convert.ToInt32(take?.ToString()), filter?.ToString() ?? null);
+        }
+
+        internal Dictionary<string, object> GetBindingDataContent(ModelBindingData bindingData)
+        {
+            return bindingData?.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, object>(bindingData?.Content?.ToObjectFromJson<Dictionary<string, object>>(), StringComparer.OrdinalIgnoreCase),
+                _ => throw new NotSupportedException($"Unexpected content-type. Currently only '{Constants.JsonContentType}' is supported.")
+            };
+        }
+
+        internal virtual async Task<object?> ToTargetTypeAsync(Type targetType, string? connection, string tableName, string? partitionKey, string? rowKey, int take, string? filter) => targetType switch
```

Comment:
There are a lot of parameters here, it might be worth using an object/model instead. i.e. instead of using a dictionary in `GetBindingDataContent` - you could deserilize into an object reference we create so that it's cleaner

Reply:
done!